das/state.go

```diff
@@ -0,0 +1,71 @@
+package das
+
+import "sync/atomic"
+
+// state collects information about the DASer process.
+type state struct {
+	// tracks whether sampling routine is running
+	isRunning uint64
+	// indicates whether the DASer catchUp routine
+	// has successfully caught up to network head.
+	isCaughtUp uint64
```

Comment:
While you can represent each state with booleans, adding more booleans per each new state is not sustainable + it's getting harder to reason about the code when you have multiple booleans and the relations between them.

I would suggest from the beginning starting with a simple state machine which can be extended going further:
> Stopped -> Running -> CatchingUp
With these 3 states, you can explain any state of a DASer.